### PR TITLE
update `torch.where` to throw error for single input

### PR DESCRIPTION
**Pull request by k223kim**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes https://github.com/Lightning-AI/lightning-thunder/issues/146. It throws a more descriptive error saying that Thunder is yet to support single input (condition only) `torch.where`.

## PR review

Anyone in the community is free to review the PR once the tests have passed.

## Did you have fun?

Loving every bit of this! 🚀

### CODE REVIEW

File: thunder/tests/opinfos.py [hunk 1]
```diff
@@ -2484,10 +2484,26 @@ def where_sample_generator(op, device, dtype, requires_grad, **kwargs):
         yield SampleInput(pred, a, b)
 
 
+def where_error_generator(op, device, dtype=torch.float32, **kwargs):
```
Comment (mruberry) [inline thunder/tests/opinfos.py:2487]: Nice error generator

File: thunder/torch/__init__.py [hunk 2]
```diff
@@ -1535,7 +1535,12 @@ def tril(a: TensorLike, /, diagonal: int = 0, *, fill_value: None | Number = Non
 
 
 @torchsymbol(torch.where, is_method=True)
-def where(pred: TensorLike, a: Number | TensorLike, b: Number | TensorLike, /) -> TensorLike:
+def where(pred: TensorLike, a: Number | TensorLike = None, b: Number | TensorLike = None, /) -> TensorLike:
```
Comment (mruberry) [inline thunder/torch/__init__.py:1538]: Update the type annotations for `a` and `b` to include `None`, like this:

```
a: None | Number | TensorLike = None, b: None | Number | TensorLike = None
```

**Review by mruberry — COMMENTED** [review]
Looks really good except for one small nit about the type annotations on `where` -- I made a comment for you review

**Comment by k223kim** [comment]
Hey @mruberry! Thanks for the review :) Just updated my code based on your comments!

**Review by mruberry — APPROVED** [review]
Cool! Thanks @k223kim!

**Comment by k223kim** [comment]
Just one more question @mruberry! It says "required statuses must pass before merging". Is there anything else I should be working on to merge this PR? It's my first PR so I am not really familiar with this process. ;)

**Comment by mruberry** [comment]
> Just one more question @mruberry! It says "required statuses must pass before merging". Is there anything else I should be working on to merge this PR? It's my first PR so I am not really familiar with this process. ;)

Nope, nothing else you need to do, @k223kim. Sorry for the confusion.

@t-vi @Borda we should probably think about clarifying the merge process somewhere

**Comment by t-vi** [comment]
Thank you @k223kim @mruberry .

@mruberry Yes, maybe you can add a little paragraph to @nikitaved (great) contributing section https://lightning-thunder.readthedocs.io/en/latest/advanced/contrib_thunder.html Thank you.

**Comment by mruberry** [comment]
> Thank you @k223kim @mruberry .
> 
> @mruberry Yes, maybe you can add a little paragraph to @nikitaved (great) contributing section [lightning-thunder.readthedocs.io/en/latest/advanced/contrib_thunder.html](https://lightning-thunder.readthedocs.io/en/latest/advanced/contrib_thunder.html) Thank you.

Not sure I'd know what to write.

**Comment by k223kim** [comment]
@mruberry 
Maybe add a small section called `When it is ready for my PR to be merged?` and add something along the lines of 'All tests are passing, at least one approval'?

**Comment by mruberry** [comment]
> @mruberry Maybe add a small section called `When it is ready for my PR to be merged?` and add something along the lines of 'All tests are passing, at least one approval'? (something like the following)
> 
> ## When it is ready for my PR to be merged?
> * [ ]  Is your code passing all tests?
> * [ ]  Did you get at least one approval?
> 
> If so, your PR is ready to be merged! 🚀

That sounds like a great start, @k223kim!

**Pull request merged**